Realm/RLMRealm.mm

```diff
@@ -780,4 +786,165 @@ - (void)detachAllEnumerators {
     _collectionEnumerators = nil;
 }
 
+#pragma mark - Async Writes
+
+- (void)writeAsyncWithBlock:(RLMAsyncWriteBlock)block completion:(RLMCompletionBlock)completion {
+    RLMRealmConfiguration *config = self.configuration;
+    
+    __block NSError *error;
+    __block BOOL didComplete;
+    
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+    __block RLMNotificationToken *token = [self addNotificationBlock:^(NSString *notification, RLMRealm *realm) {
```

Comment:
This shouldn't need to be a `__block` parameter since the value of the pointer isn't being mutated inside the block, right?